infra/src/test/kotlin/fi/fta/geoviite/api/tracklayout/v1/ExtTrackLayoutTestApiService.kt

```diff
@@ -14,8 +15,33 @@ class ExtTrackLayoutTestApiService(mockMvc: MockMvc) {
     val testApiMapper = jacksonObjectMapper().apply { setSerializationInclusion(JsonInclude.Include.NON_NULL) }
     val testApiConnection = TestApi(testApiMapper, mockMvc)
 
+    val trackLayoutVersion =
+        AssetApi<Uuid<Publication>, ExtTestTrackLayoutVersionV1, Nothing, Nothing>(
+            assetUrl = { uuid -> "/geoviite/paikannuspohja/v1/versiot/${uuid}" },
+            assetClazz = ExtTestTrackLayoutVersionV1::class,
+        )
+
+    val trackLayoutVersionLatest =
+        AssetCollectionApi<ExtTestTrackLayoutVersionV1, Nothing>(
+            assetCollectionUrl = { "/geoviite/paikannuspohja/v1/versiot/uusin" },
+            assetCollectionClazz = ExtTestTrackLayoutVersionV1::class,
+        )
+
+    val trackLayoutVersionCollection =
+        AssetCollectionApi(
+            assetCollectionUrl = { "/geoviite/paikannuspohja/v1/versiot" },
+            assetCollectionClazz = ExtTestTrackLayoutVersionCollectionResponseV1::class,
+            modifiedAssetCollectionUrl = { "/geoviite/paikannuspohja/v1/versiot/muutokset" },
+            modifiedAssetCollectionClazz = ExtTestTrackLayoutVersionCollectionResponseV1::class,
+        )
+
     val locationTracks =
-        AssetApi(
+        AssetApi<
+            Oid<*>,
```

Comment:
Lisäsin näihin api-määrityksiin id-tyypityksen sillä versioilla ei ole oid vaan uuid. Tätäkin pitäisi oikeastaan miettiä nyt uusiksi ja hahmottelin jo vähän jotain tällaista:
API tyyppeinä simpleApi (ne joihin ei mene id:tä) / assetApi (ne joihin menee) + modification-versiot molemmille. Sitten kompositiolla rakennettu yhdistelmä johon voi laittaa erilliset osa-alueet (yksittäinen/lista/geometria/muutokset/listamuutokset/jne).

Tuolla pääsisi tarpeesta määritellä kaikkiin käsitteisiin noita Nothing, Nothing juttuja ja käyttäjän kannalta se näyttäisi luultavasti melko samalta. Tuokin remppalistalle mietittäväksi kun vähän vielä kirjoittelen lisää näitä.